torchtune/training/quantization.py

```diff
@@ -219,3 +223,22 @@ def swap_lora_linear_with_qat(
                 activation_qat_config,
                 weight_qat_config,
             )
+
+
+def convert_to_float8_training(model: torch.nn.Module) -> torch.nn.Module:
+    """
+    Prepare the model for float8 training by swapping all `torch.nn.Linear` with `Float8Linear`.
+    """
+    print("doing fp8 quantized training")
+    fp8_config = Float8LinearConfig(
+        enable_fsdp_float8_all_gather=True,
+        force_recompute_fp8_weight_in_bwd=True,
+        cast_config_input=CastConfig(scaling_type=ScalingType.DYNAMIC),
+        cast_config_weight=CastConfig(scaling_type=ScalingType.DYNAMIC),
+        cast_config_grad_output=CastConfig(scaling_type=ScalingType.DYNAMIC),
+    )
```

Comment:
it would be good to also enable the rowwise scaling recipes here for increased accuracy! `Float8LinearConfig.from_recipe_name(...)`, https://github.com/pytorch/ao/blob/c9b1490971ab3b95c20f076e8dd3a679e27883dc/torchao/float8/config.py#L114

Note that only the all-tensorwise recipe needs the custom FSDP and TP things.